### PR TITLE
add allowloop to repeat dialog and test

### PR DESCRIPTION
**Pull request by xieofxie**
Close https://github.com/microsoft/botbuilder-dotnet/issues/3742

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/RepeatDialog.cs [hunk 1]
```diff
@@ -26,6 +26,18 @@ public RepeatDialog(object options = null, [CallerFilePath] string callerPath =
...
+        /// Gets or sets an optional expression which if is true will allow loop of the repeated dialog.
+        /// </summary>
+        /// <example>
+        /// "user.age > 18".
+        /// </example>
+        /// <value>
+        /// A boolean expression.
+        /// </value>
+        [JsonProperty("allowLoop")]
+        public BoolExpression AllowLoop { get; set; }
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/RepeatDialog.cs:39]: >AllowLoop [](start = 30, length = 9)

You need to update Microsoft.RepeatDialog.schema with this property.

Comment (xieofxie) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/RepeatDialog.cs:39]: Updated~

**Comment by xieofxie** [comment]
OK, it seems tests passed but no result for coverage..

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/RepeatDialog.cs [hunk 2]
```diff
@@ -58,10 +70,16 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             var repeatedIds = dc.State.GetValue<List<string>>(TurnPath.RepeatedIds, () => new List<string>());
             if (repeatedIds.Contains(targetDialogId))
             {
-                throw new ArgumentException($"Recursive loop detected, {targetDialogId} cannot be repeated twice in one turn.");
+                if (this.AllowLoop == null || this.AllowLoop.GetValue(dc.State) == false)
+                {
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/RepeatDialog.cs:74]: I would move this conidtion around the entire repeatedIds section, no need to track ids if you aren't using them.

**Pull request merged**